tests/conftest.py

```diff
@@ -0,0 +1,127 @@
+"""Define test configuations."""
+
+import random
+from pathlib import Path
+from shutil import rmtree
+from typing import Generator
+
+import numpy as np
+import pytest
+from hydra import compose, initialize
+from omegaconf import DictConfig
+
+from zebra_puzzles.evaluation import evaluate_all
+from zebra_puzzles.pipeline import build_dataset
+
+initialize(config_path="../config", version_base=None)
+
+
+# Test configurations for pytest
+# Each set of parameters represents a different configuration for the tests.
+@pytest.fixture(
+    scope="session", params=[(1, 2, 0), (2, 2, 0), (1, 2, 2), (2, 2, 2), (2, 0, 0)]
+)
+def config(request) -> Generator[DictConfig, None, None]:
+    """Hydra configuration.
+
+    This fixture yields a Hydra configuration object for the tests. It uses params to create multiple configurations.
+    """
+    # Set the seeds of random and numpy
+
+    random.seed(42)
+    np.random.seed(42)
+
+    yield compose(
+        config_name="config",
+        overrides=[
+            f"n_puzzles={request.param[0]}",
+            "n_objects=3",
+            "n_attributes=3",
+            f"n_red_herring_clues={request.param[1]}",
+            f"n_red_herring_clues_evaluated={request.param[2]}",
+            "data_folder=tests/test_data",
+            "model=gpt-4o-mini",
+        ],
+    )
```

Comment:
You could make it a bit more transparent:

```suggestion
    n_puzzles, n_red_herring_clues, n_red_herring_clues_evaluated = request.param
    yield compose(
        config_name="config",
        overrides=[
            f"n_puzzles={n_puzzles}",
            "n_objects=3",
            "n_attributes=3",
            f"n_red_herring_clues={n_red_herring_clues}",
            f"n_red_herring_clues_evaluated={n_red_herring_clues_evaluated}",
            "data_folder=tests/test_data",
            "model=gpt-4o-mini",
        ],
    )
```